mbed/source/main.cpp

```diff
@@ -0,0 +1,423 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2006-2021 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cctype>
+#include <chrono>
+#include <events/mbed_events.h>
+
+#include "ble/BLE.h"
+#include "pretty_printer.h"
+#include "bt_test_state.hpp"
+
+/** This program enters different BLE states according to operator input, allowing power consumption to be
+ * measured.
+ */
+
+using namespace std::literals::chrono_literals;
+
+static const char DEVICE_NAME[] = "Power Consumption";
+static const uint16_t MAX_ADVERTISING_PAYLOAD_SIZE = 50;
+
+events::EventQueue event_queue;
+
+class PowerConsumptionTest : public ble::Gap::EventHandler {
+public:
+    PowerConsumptionTest(BLE& ble, events::EventQueue& event_queue) :
```

Comment:
To avoid shadowing the global variable:

```suggestion
    PowerConsumptionTest(BLE& ble, events::EventQueue& eq) :
```